superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -481,6 +481,30 @@ export default class ResultSet extends React.PureComponent<
     return <div className="noControls" />;
   }
 
+  rowsReturned() {
+    const { query } = this.props;
+    let limitWarning = null;
+    if (query.results?.displayLimitReached) {
+      limitWarning = <Icon className="ReturnedRowsImage" name="warning" />;
+    }
+    return (
+      <div className="ReturnedRows">
+        {limitWarning}
```

Comment:
a bit of a nit, but instead of making `limitWarning` a react component or null and using it as a boolean, it would be better to use `displayLimitReached` as the boolean in both cases and have limitWarning always be the icon. 

It would read `{displayLimitReached && limitWarning}` for example and the reader would know that it only shows the limitWarning if the displayLimit is reached, rather than always showing the warning, which will be null if the display limit isn't reached. It's a small nuance, but will help.